src/vorta/borg/borg_thread.py

```diff
@@ -135,40 +139,74 @@ def run(self):
                                   profile=self.params.get('profile_name', None)
                                   )
         log_entry.save()
+        logger.info('Running command %s', ' '.join(self.cmd))
 
-        self.process = Popen(self.cmd, stdout=PIPE, stderr=PIPE, bufsize=1,
-                             universal_newlines=True, env=self.env, preexec_fn=os.setsid)
+        p = Popen(self.cmd, stdout=PIPE, stderr=PIPE, bufsize=1, universal_newlines=True,
+                  env=self.env, cwd=self.cwd, preexec_fn=os.setsid)
+        self.process = p
 
-        for line in iter(self.process.stderr.readline, ''):
+        # Prevent blocking. via https://stackoverflow.com/a/7730201/3983708
+
+        # Helper function to add the O_NONBLOCK flag to a file descriptor
+        def make_async(fd):
+            fcntl.fcntl(fd, fcntl.F_SETFL, fcntl.fcntl(fd, fcntl.F_GETFL) | os.O_NONBLOCK)
```

Comment:
https://docs.python.org/3/library/os.html#os.set_blocking

Reply:
So this would be a shortcut instead of using this function?

Reply:
yup.

Reply:
Good tip. This made it a lot [simpler](https://github.com/borgbase/vorta/pull/63/files#diff-74bac960beabf99e83633ac112450502). Thanks.